pkgs/applications/misc/comfyui/default.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, python311
+, fetchPypi
+, writers
+, writeTextFile
+, fetchFromGitHub
+, stdenv
+, pkgs
+, torch ? python311.pkgs.torch
+, modelsPath ? "/opt/comfyui/models"
+, inputPath ? "/opt/comfyui/input"
+, outputPath ? "/opt/comfyui/output"
+, customNodesPath ? "/opt/comfyui/custom_nodes"
+}:
+
+let
+  modelPathsFile = writeTextFile {
+    name = "model_paths.yaml";
+    text = (lib.generators.toYAML {} {
+      comfyui = {
+        base_path = modelsPath;
+        checkpoints = "${modelsPath}/checkpoints";
+        clip = "${modelsPath}/clip";
+        clip_vision = "${modelsPath}/clip_vision";
+        configs = "${modelsPath}/configs";
+        controlnet = "${modelsPath}/controlnet";
+        embeddings = "${modelsPath}/embeddings";
+        loras = "${modelsPath}/loras";
+        upscale_models= "${modelsPath}/upscale_models";
+        vae = "${modelsPath}/vae";
+        vae_approx = "${modelsPath}/vae_approx";
+      };
+    });
+  };
+
+  executable = writers.writeDashBin "comfyui" ''
+    cd $out && ${python311}/bin/python comfyui "$@"
+  '';
+in stdenv.mkDerivation rec {
+  pname = "comfyui";
+  version = "8a451234b3090db488fbee9740a5f6be2f989253";
```

Comment:
If upstream does not have tagged releases the version should be called `unstable-yyyy-mm-dd`, where the yyy-mm-dd is the date of the commit you checked out.